int64 num
---
int64 class_num

